src/controller/python/README.md

```diff
@@ -0,0 +1,162 @@
+# CHIP Device Controller - Python Binding & Command line interface
+
+## Overview
+
+Chip Device Controller is a general library to control devices and manage
+commissioner functionality, it has Python, Java (for Android) and ObjC (for iOS)
+bindings for building tests and CHIP mobile applications. This document will
+focus on its python binding interface on Linux and mac os.
+
+## Checkout / Build / Install
+
+> Note: The CHIP can be built on Linux (amd64 / aarch64) and macOS
+
+1. Clone Project CHIP repo.
+
+```
+git clone https://github.com/project-chip/connectedhomeip.git
+```
+
+2. Build CHIP
+
+Follow [BUILDING.md](/docs/BUILDING.md) to build CHIP on your platform.
+
+3. Install Chip Device Controller
+
+**For Linux**
+
+```
+cd out/debug/linux_arm64_clang/controller/python
+sudo pip3 install chip-0.0-cp38-cp38-linux_aarch64.whl
+```
+
+> Note: for linux, sudo is necessary since it need to interact with bluetoothd
+> via dbus with sudo permission
+
+**For macOS**
+
+```
+cd out/debug/mac_x64_clang/controller/python
+pip3 install chip-0.0-cp38-cp38-mac_aarch64.whl
+```
+
+## BLE Virtualization on Linux (Optional)
+
+If we would like to setup virtual BLE central and peripheral in the same
+machine, then Bluez setup
+
+```
+cd third_party/bluez/repo
+./bootstrap
+third_party/bluez/repo/configure --prefix=/usr --mandir=/usr/share/man --sysconfdir=/etc --localstatedir=/var --enable-experimental --with-systemdsystemunitdir=/lib/systemd/system --with-systemduserunitdir=/usr/lib/systemd --enable-deprecated --enable-testing --enable-tools
```

Comment:
What's wrong with the Ubuntu bluez?

Reply:
currently we use bluez 5.55, couple of bluez upstream fix have been submitted to upstream by teams in the past year. Also the default cannot work, we need to enable bluetoothd with experimental flag since some necessary bluez feature are under experimental.

Reply:
in addition, btvirt tool needs to be generated from source, since standard distribution don't have it.

Reply:
According to my tests, its ok to use default bluez shipped with the disto if it is 5.55 (ubuntu 20.10 does this), however, if it is 5.53 (ubuntu 20.04 lts) we may need to build bluez from the source.
Anyway, btvirt requires to build from the source.

Reply:
i think it is fine to put it as optional if their os has 5.55